### PR TITLE
Feature/si 2198 manufacturer factory reset

### PR DESCRIPTION
**Pull request by LorranSutter**
This is PR is supposed to merge into dev. Github is having some issues and I can't point to dev for now.

### CODE REVIEW

File: contracts/implementations/nodes/AftermarketDevice.sol [hunk 1]
```diff
@@ -518,6 +522,56 @@ contract AftermarketDevice is
         _setInfos(tokenId, attrInfo);
     }
 
+    /**
+     * @notice Set the device address of a list of aftermarket devices
+     * Caller must be the owner of the aftermarket device's parent manufacturer or an authorized address
+     * The manufacturer node owner must grant the MANUFACTURER_FACTORY_RESET_PRIVILEGE privilege to the authorized address
+     * @param adIdAddrs List of deviceId-deviceAddress pairs to be set
+     */
+    function setAftermarketDeviceAddressByManufacturerBatch(
```
Comment (elffjs) [inline contracts/implementations/nodes/AftermarketDevice.sol:531]: Maybe we should call it `resetAftermarketDeviceAddressByManufacturerBatch`? Just to match the event?

Comment (LorranSutter) [reply contracts/implementations/nodes/AftermarketDevice.sol:531]: Or we could rename de event.

File: contracts/implementations/nodes/AftermarketDevice.sol [hunk 2]
```diff
@@ -518,6 +522,56 @@ contract AftermarketDevice is
...
+        );
+
+        uint256 tokenId;
+        address newDeviceAddress;
+        uint256 manufacturerParentNode;
+        for (uint256 i = 0; i < adIdAddrs.length; i++) {
+            tokenId = adIdAddrs[i].aftermarketDeviceNodeId;
+            newDeviceAddress = adIdAddrs[i].deviceAddress;
+            manufacturerParentNode = ns
+            .nodes[adIdProxyAddress][tokenId].parentNode;
```
Comment (elffjs) [inline contracts/implementations/nodes/AftermarketDevice.sol:551]: I'm surprised it's formatting like this.

**Review by elffjs — APPROVED** [review]

**Comment by elffjs** [comment]
Let's merge?

**Pull request merged**